### ISSUE TITLE
Error parsing TAP Stream: Invalid indentation

### ISSUE THREAD

**Issue by epaminond** [tupilabs/tap4j#47]
Getting this error when trying to parse TAP-results through [Jenkins TAP-plugin](https://wiki.jenkins-ci.org/display/JENKINS/TAP+Plugin) (latest 2.1 version). TAP-file is generated via `ember test --silent` (which uses [testem](https://github.com/testem/testem) under the hood) and has 4 spaces indentation.

Is it an issue with tap4j or with file?

Using tap4j-4.2.1 release.

Error trace:
```
org.tap4j.parser.ParserException: Error parsing TAP Stream: Invalid indentation. Check your TAP Stream. Line:     ---
	at org.tap4j.parser.Tap13Parser.parseTapStream(Tap13Parser.java:230)
	at org.tap4j.parser.Tap13Parser.parseFile(Tap13Parser.java:193)
	at org.tap4j.plugin.TapParser.parse(TapParser.java:167)
	at org.tap4j.plugin.TapPublisher.loadResults(TapPublisher.java:540)
	at org.tap4j.plugin.TapPublisher.performImpl(TapPublisher.java:412)
	at org.tap4j.plugin.TapPublisher.perform(TapPublisher.java:371)
```

The TAP-file example is as follows:
```
skip 1 PhantomJS 2.1 - Acceptance | company settings: it shows current settings
    ---
        Log: |
            { type: 'warn',
              text: '\'DEPRECATION: Ember.K is deprecated in favor of defining a function inline. [deprecation id: ember-metal.ember-k] See http://emberjs.com/deprecations/v2.x#toc_code-ember-k-code for more details.\\n    http://localhost:7357/assets/vendor.js:17193:15\\n    raiseOnDeprecation@http://localhost:7357/assets/vendor.js:17112:17\\n    http://localhost:7357/assets/vendor.js:17193:15\\n    invoke@http://localhost:7357/assets/vendor.js:17209:21\\n    deprecate@http://localhost:7357/assets/vendor.js:17177:37\\n    deprecate@http://localhost:7357/assets/vendor.js:29019:42\\n    get@http://localhost:7357/assets/vendor.js:54526:28\\n    http://localhost:7357/assets/vendor.js:172502:30\\n    exports@http://localhost:7357/assets/vendor.js:140:37\\n    _reify@http://localhost:7357/assets/vendor.js:173:66\\n    reify@http://localhost:7357/assets/vendor.js:159:33\\n    exports@http://localhost:7357/assets/vendor.js:138:15\\n    _reify@http://localhost:7357/assets/vendor.js:173:66\\n    reify@http://localhost:7357/assets/vendor.js:159:33\\n    exports@http://localhost:7357/assets/vendor.js:138:15\\n    requireModule@http://localhost:7357/assets/vendor.js:32:25\\n    resolveInitializer@http://localhost:7357/assets/vendor.js:143623:25\\n    registerInitializers@http://localhost:7357/assets/vendor.js:143636:41\\n    http://localhost:7357/assets/vendor.js:143662:25\\n    http://localhost:7357/assets/metis.js:284:41\\n    exports@http://localhost:7357/assets/vendor.js:140:37\\n    _reify@http://localhost:7357/assets/vendor.js:173:66\\n    reify@http://localhost:7357/assets/vendor.js:159:33\\n    exports@http://localhost:7357/assets/vendor.js:138:15\\n    _reify@http://localhost:7357/assets/vendor.js:173:66\\n    reify@http://localhost:7357/assets/vendor.js:159:33\\n    exports@http://localhost:7357/assets/vendor.js:138:15\\n    requireModule@http://localhost:7357/assets/vendor.js:32:25\\n    http://localhost:7357/assets/test-support.js:5319:18\\n    require@http://localhost:7357/assets/test-support.js:5309:32\\n    loadModules@http://localhost:7357/assets/test-support.js:5301:23\\n    load@http://localhost:7357/assets/test-support.js:5252:37\\n    http://localhost:7357/assets/test-support.js:5173:22\'\n' }
    ...
skip 2 PhantomJS 2.1 - Acceptance | company settings: it allows to change settings
ok 3 PhantomJS 2.1 - JSCS - acceptance/company-settings-test.js: should pass jscs
<skipped>
ok 1087 PhantomJS 2.1 - JSCS - validators/unique-vat.js: should pass jscs
ok 1088 PhantomJS 2.1 - ESLint - validators/unique-vat.js: should pass ESLint

1..1088
# tests 1088
# pass  1078
# skip  10
# fail  0

# ok

```

**Comment by kinow** [comment]
Tested on instanttap, and it said the TAP stream was valid. Tested just the YAMLish block in another online service, and it also passed the validation tests.

Will look into issue during the next development cycle, likely something changed recently. instanttap is probably using an older version of tap4j.

**Comment by epaminond** [comment]
That was really quick. :)

**Comment by epaminond** [comment]
@kinow , are there any updates on this?

**Comment by H1D** [comment]
Found dirty workaround. Just start you tests like so:
`echo 'TAP version 13' && ember test --silent`

**Comment by japod** [comment]
I think the issue should be solved by https://github.com/japod/tap4j/commit/2c61eb08eb99ed562d06f7f6e2158179e6736268
Not sure if it could be merged as is (as my branch probably diverged in the meantime) to the main trunk. I want to spend some time on it and try to contribute all my changes back to the original repository.

**Comment by japod** [comment]
Hm, the above commit does not really solve the issue with the TAP file provided. Will see what i could do, but do not have any solution at the moment.

**Comment by japod** [comment]
It is written in the spec, that a yaml block is interpreted as a yaml block if it follows immediately after a test line:

YAML blocks
If the test line is immediately followed by an indented block beginning with /^\s+---/ and ending with /^\s+.../ that block will be interpreted as an inline YAML document.

The above test case does not have any test line immediately preceding the yaml block. I attempted to fix this in https://github.com/tupilabs/tap4j/pull/59/

**Comment by kinow** [comment]
>The above test case does not have any test line immediately preceding the yaml block, but our parser still tried to interpret the text as yaml instead of ignoring it. I attempted to fix this in #59

Thanks for looking it at in the spec, and for the explanation. Let me check #59 and see if we can merge/release.

Thanks @japod

**Comment by kinow** [comment]
@japod as the related PR was merged, can we close this one? Will roll a release with the current code this weekend.

**Comment by japod** [comment]
Sure, closing it. Many thanks for the quick responses and for taking care of the release!

**Comment by H1D** [comment]
Wow! Thanks for fast reaction!

**Comment by kinow** [comment]
All props to @japod (thanks) :clap: :clap: and thanks everyone for the patience. Cutting release now, should be available from Maven central in the next hours.

**Comment by H1D** [comment]
@kinow what is release cycle for Jenkins TAP plugin?

Can we have new release – https://github.com/jenkinsci/tap-plugin ?

**Comment by kinow** [comment]
@japod updated the version of tap4j there too (thanks!). Probably a new release in a few days.

https://github.com/jenkinsci/tap-plugin/pull/24

**Comment by kinow** [comment]
Unless we try to release master tomorrow and then cut another minor release right after that. WDYT @japod?

**Comment by japod** [comment]
I have no problem with releasing the master tomorrow. It would be maybe good to still include https://github.com/jenkinsci/tap-plugin/pull/25 in the release if you are o.k. with that (stripping corrupted YAML should be configurable).

The two additional updates that i have in the pipeline locally (and i still need to adjust them to the master branch in the coming days) are:
- fixed graph rendering
- a few new dashboard columns related to TAP test results

**Comment by H1D** [comment]
https://github.com/jenkinsci/tap-plugin/pull/25 would be nice to have with new tap-plugin release

**Comment by kinow** [comment]
Released 2.3, updated the [Wiki](https://wiki.jenkins.io/display/JENKINS/TAP+Plugin) as well.

Thanks @japod, @H1D, et al!
